File: java/interviewPratice/RotateImage.java
public final class RotateImage {

    int[][] rotateImage(int[][] a) {
        int[][] out = new int[a.length][a[0].length];
        int i = a.length - 1;
        for(int x = 0; x < out[0].length; x++, i--) {
            for(int y = 0; y < out.length; y++) {
                out[y][x] = a[i][y];
            }
        }
        return out;
    }
}

File: InterviewPratice/Javascript/rotateImage.js
const rotateImage = (a) => {
    let out = [];
    
    for(let i = 0; i < a.length; i++) out.push(new Array(a[0].length));
        
    for(let x = 0, i = a.length - 1; x < a[0].length; x++, i--) {
        for(let y = 0; y < a.length; y++) out[y][x] = a[i][y];
    }
    
    return out;
}